nvflare/apis/impl/controller.py

```diff
@@ -144,3 +145,6 @@ def cancel_task(
 
     def cancel_all_tasks(self, completion_status=TaskCompletionStatus.CANCELLED, fl_ctx: Optional[FLContext] = None):
         self.communicator.cancel_all_tasks(completion_status, fl_ctx)
+
+    def handle_event(self, event_type: str, fl_ctx: FLContext):
```

Comment:
No, this would make all subclasses of Controller have to call super().handle_event.
A better approach is to add the communicator to the handler_list. See build_component() in server_json_config.py in nvflare/private/fed/server how to do that. Basically once you create the "communicator" object, simply call self.handlers.append(communicator). Of course the communicator must be a FLComponent.